### ISSUE TITLE
Print the line from where the issue is raised

### ISSUE THREAD

**Issue by NicolasMICAUX** [DontShaveTheYak/deep-ast#34]
Hi :)
Great lib!
In the [example with exceptions](https://github.com/DontShaveTheYak/deep-ast#example-to-get-all-exceptions), do you see a way to report the line where the issue is raised?
The use case would be to understand where the issue is raised, and if one should try to handle it or not. Having the context of the "raise" line makes it easier to understand what is the error (especially with often used exception like TypeError).

example:
{'ValueError': 'Line 13,
'TypeError': 'Line 17'}

I'm not very familiar with AST but seems that one can get the attributes 'lineno', 'col_offset', 'end_lineno', 'end_col_offset' for which node. We still need to trace parents to get like where in the file is this from.

If you have an idea of how to do this, I can implement it. :)

**Comment by shadycuz** [comment]
@NicolasMICAUX 

I think you could modify this function:
```python
def _add_exception(self, name: str):
        self.raw_exceptions.append(name)

        if name not in self.found_exceptions:
            self.found_exceptions.append(name)
```
Make it take a node instead of name. 

Then create a function that prints all the Exceptions in the format you expect. 

Like:
```python
def print_exceptions(self):
  for exception in self.found_exceptions:
    print(f"Found {exception.name} on line {exception.lineno}`)
```

I think something like that would work?

**Comment by NicolasMICAUX** [comment]
Yes that's kind of what I've tried. But I've never been able to recover the file from which the line is from. So when analyzing multi file code, it would not work :(

**Comment by shadycuz** [comment]
Do you need the path to the file or the name of it?

The path might not be that helpful as it's usually pretty long.

**Comment by NicolasMICAUX** [comment]
Ideally the path when one want to find exceptions raised in a lot of submodules, for example i was using it to find errors raised by a complex dataloader.
But true that having the name of the file is better than nothing.
